test/api_integration/apis/saved_objects/bulk_update.js

```diff
@@ -0,0 +1,288 @@
+        await supertest
+          .put(`/api/saved_objects/_bulk_update`)
+          .send([
+            {
+              type: 'visualization',
+              id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+              attributes: {
+                title: 'An existing visualization'
+              }
+            },
+            {
+              type: 'dashboard',
+              id: 'be3733a0-9efe-11e7-acb3-3dab96693fab',
+              attributes: {
+                title: 'An existing dashboard'
+              }
+            },
+          ])
+          .expect(200)
+          .then(resp => {
+
+            const { saved_objects: [ firstObject, secondObject ] } = resp.body;
+
+            // loose ISO8601 UTC time with milliseconds validation
+            expect(firstObject).to.have.property('updated_at').match(/^[\d-]{10}T[\d:\.]{12}Z$/);
+            expect(_.omit(firstObject, ['updated_at'])).to.eql({
+              id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+              type: 'visualization',
+              version: 'WzgsMV0=',
+              attributes: {
+                title: 'An existing visualization',
+              },
+            });
+
+            expect(secondObject).to.have.property('updated_at').match(/^[\d-]{10}T[\d:\.]{12}Z$/);
+            expect(_.omit(secondObject, ['updated_at'])).to.eql({
+              id: 'be3733a0-9efe-11e7-acb3-3dab96693fab',
+              type: 'dashboard',
+              version: 'WzksMV0=',
+              attributes: {
+                title: 'An existing dashboard',
+              },
+            });
+          });
+      });
+
+      it('does not pass references if omitted', async () => {
+        const { body: { saved_objects: [ visObject, dashObject ] } } = await supertest
+          .post(`/api/saved_objects/_bulk_get`)
+          .send([
+            {
+              type: 'visualization',
+              id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+            },
+            {
+              type: 'dashboard',
+              id: 'be3733a0-9efe-11e7-acb3-3dab96693fab',
+            }
+          ]);
+
+        await supertest
+          .put(`/api/saved_objects/_bulk_update`)
+          .send([
+            {
+              type: 'visualization',
+              id: 'dd7caf20-9efd-11e7-acb3-3dab96693fab',
+              attributes: {
+                title: 'Changed title but nothing else'
+              },
+              options: {
+                version: visObject.version
+              }
+            },
+            {
+              type: 'dashboard',
+              id: 'be3733a0-9efe-11e7-acb3-3dab96693fab',
+              attributes: {
+                title: 'Changed title and references'
+              },
+              options: {
+                version: dashObject.version,
+                references: [{ id: 'foo', name: 'Foo', type: 'visualization' }]
+              }
+            },
+          ])
+          .expect(200)
+          .then(async (resp) => {
```

Comment:
nit: I think it would make it a bit more readable if we didn't nest the rest of the test under the `.then()` but rather used await to get the response.

Reply:
Fair enough. :)